internal/codespaces/connection/tunnels_api_server_mock.go

```diff
@@ -0,0 +1,444 @@
+package connection
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"regexp"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/gorilla/websocket"
+	"github.com/microsoft/dev-tunnels/go/tunnels"
+	tunnelssh "github.com/microsoft/dev-tunnels/go/tunnels/ssh"
+	"github.com/microsoft/dev-tunnels/go/tunnels/ssh/messages"
+	"golang.org/x/crypto/ssh"
+)
+
+func NewHttpClient() (*http.Client, error) {
```

Comment:
Small nit but do you think we can add some indicator to this name to suggest it is a testing http client?

```suggestion
func NewMockHttpClient() (*http.Client, error) {
```

Reply:
Good idea!

Reply:
There were also quite a few public funcs and types that didn't need to be so I went ahead and made them private so callers don't accidentally use them.